source/materials/XenonGasProperties.cc

```diff
@@ -239,6 +239,15 @@ namespace nexus {
 
 
 
+  void XenonGasProperties::Scintillation
```

Comment:
Why are there two methods `void XenonGasProperties::Scintillation` and also `void XenonLiquidProperties::Scintillation`?

Reply:
Because the scintillation spectrum is different in gaseous and liquid xenon.

Reply:
No, but there are in total four methods:
- `void XenonGasProperties::Scintillation`
- `void XenonGasProperties::Scintillation`
- `void XenonLiquidProperties::Scintillation`
- `void XenonLiquidProperties::Scintillation`

Reply:
I see what you mean, that we should delete the old signature 

```
Scintillation
  (G4int entries, G4double* energy, G4double* intensity)
```

right? Will do.

Reply:
However, we still have `GAr` which has the old format. I either would keep both signatures or update the GAr property.

Reply:
Yes, thanks!

Reply:
Hint: those are C-arrays in that signature!

Reply:
> But will it work?

Yes, in fact there are three functions with the same name, but they have different signatures and correct one will be used in the right place, according to the arguments passed.

Reply:
> Hint: those are C-arrays in that signature!

What do you mean @jacg ?

Reply:
Signatures like 

``` c++
fn(int N, double* this, double* that);
```

(containing pointers to the start of contiguously-allocated chunks of homogeneous memory, and an accompanying int to tell us how much memory the pointer points to, AKA C-arrays) are what you would write in C. In C++ *you should not use C-arrays*! (For a plethora of well-known and oft-repeated reasons that I do not wish to rehash here.) C++ admits them (and makes `std::vector` trivially compatible with them), in case you want to communicate with a legacy C-API, but you should not write your own functions with such signatures (unless it's a callback you're going to pass to C).

The fact that Geant4 perpetuates interfaces that have been deprecated for more than a quarter of a century, is no excuse for us to do the same (rule of thumb: if G4 does it, it's probably a bad idea (and I'm only half-joking)) ... and even Geant4 has started to wake up to the fact in 10.7!

Remove that signature, hunt down and eradicate others like it that might be lurking in the code (hope and pray that no clients have already started using them), and don't be tempted to write any more like it!